### ISSUE TITLE
🙏 Badge request for SDMAN! 🥇 

### ISSUE THREAD

**Issue by adriens** [badges/shields#7581]
# :clipboard: **Description**

This badge is dedicated to [SDKMAN!](https://sdkman.io/) so people can share that : 

- 1️⃣  Their software is distributed through `sdk`
- 2️⃣ The latest version of the released software

A clear and concise description of the new badge.

- Which service is this badge for e.g: GitHub, Travis CI : **[SDKMAN!](https://sdkman.io/)**
- What sort of information should this badge show? : **current version**

# 📷  Static example

https://img.shields.io/badge/SDKMAN!-16.16.11-blue

![image](https://user-images.githubusercontent.com/5235127/152878056-61ee843c-9a08-40a2-93bf-6510e9845cf7.png)

# :link: **Data**

Where can we get the data from?

- Is there a public API? [Yes](https://github.com/sdkman/sdkman-candidates/blob/master/features/default.feature)
- Does the API requires an API key? **Nope**
- Link to the API documentation. [here](https://github.com/sdkman/sdkman-candidates/blob/master/features/default.feature)

# :microphone: **Motivation**

On software homepage, it would help people know that : 

- 1️⃣  Core software is distributed through [SDKMAN!](https://sdkman.io/)... which makes things very easy
- 2️⃣  The target available version

 ## ⚡ Specific use case?

Embed shield : 

- 1️⃣  On GH README homepage
- 2️⃣  On [official sdks list](https://sdkman.io/sdks)

**Comment by chris48s** [comment]
Just to summarise this a bit
Example valid API call: https://api.sdkman.io/2/candidates/default/scala
Example invalid API call: https://api.sdkman.io/2/candidates/default/does-not-exist

While this doesn't appear to be a publicly documented API, we do have one of the project maintainers suggesting its use in https://github.com/sdkman/sdkman-cli/issues/1045

**Comment by adriens** [comment]
Thanks for your feedback @chris48s . The approach is being reviewed on [the sdkman side](https://github.com/sdkman/sdkman-cli/issues/1045#issuecomment-1033022411)

**Comment by calebcartwright** [comment]
While this project does certainly fall into the category of the types of developer/technologist-centric service and has some good adoption and traction, I was also wondering about the documentation.

One of the things we've called out explicitly in [our guidelines for new badges](https://github.com/badges/shields/blob/master/CONTRIBUTING.md#badge-guidelines) is:

> Badges should not obtain data from undocumented or reverse-engineered API endpoints.

Due to issues we've had in the past. I think this would be a reasonable badge to include, but I'd really like to see the documentation formalized a bit first. FWIW something as simple as a hosted swagger page has been sufficient in many prior cases

**Comment by adriens** [comment]
> Due to issues we've had in the past. I think this would be a reasonable badge to include, but I'd really like to see the documentation formalized a bit first. 

Sure 🙏 

> FWIW something as simple as a hosted swagger page has been sufficient in many prior cases

❤️

**Comment by adriens** [comment]
✅ [Commented on the SDKMAN!](https://github.com/sdkman/sdkman-cli/issues/1045#issuecomment-1033259603) side

**Comment by adriens** [comment]
cf https://github.com/sdkman/sdkman-cli/issues/1045#issuecomment-1033663080 : 

![image](https://user-images.githubusercontent.com/5235127/153294874-4a46ceea-d1c6-4e16-80e3-af7d6a82f442.png)

**Comment by calebcartwright** [comment]
Thanks for facilitating the cross-thread comms @adriens!

I confess I'm a little torn on whether we should proceed with the API in its current state or wait till it's more official. Curious what others think

**Comment by chris48s** [comment]
The reason we added the line

> Badges should not obtain data from undocumented or reverse-engineered API endpoints.

was because we had some situations where a contributor worked out that a project had an API by inspecting the frontend code for their website or whatever, added a badge that calls it, and then we ended up relying on an API that wasn't really intended for public use. This then either annoyed the maintainers of the company/project behind it or we had no reasonable expectation of stability/compatibility over time (because devs maintaining the the API were making the assumption the only thing consuming it was their own frontend code) and stuff kept breaking.

I'm less worried about someone adding a badge that calls this API if the maintainers of the project are actively suggesting we use it for this purpose, even if it lacks formal docs.

**Comment by calebcartwright** [comment]
Yeah that sounds fair to me. I think as a related, but separate, followup it'd be worth considering augmenting the text with a clause to that effect

**Comment by adriens** [comment]
@calebcartwright : 

>  we should proceed with the API in its current state or wait till it's more official

i guess the _"wait till it's more official"_ is the most appropriated option 🤔

**Comment by calebcartwright** [comment]
@adriens - we're actually saying the opposite, and that we're fine with proceeding now given what the SDMAN maintainers have indicated

**Comment by adriens** [comment]
WOW, that sounds great 👏

**Comment by mariday** [comment]
I would like to give this issue a try!

**Comment by Shaz666** [comment]
Hey this looks simple, can I work on it? I am a first-time contributor
